samcli/lib/remote_invoke/stepfunctions_invoke_executors.py

```diff
@@ -0,0 +1,119 @@
+import logging
+import time
+from typing import Any
+
+from botocore.exceptions import ClientError, ParamValidationError
+
+from samcli.lib.remote_invoke.exceptions import (
+    ErrorBotoApiCallException,
+    InvalideBotoResponseException,
+    InvalidResourceBotoParameterException,
+)
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeIterableResponseType,
+    RemoteInvokeLogOutput,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeRequestResponseMapper,
+    RemoteInvokeResponse,
+)
+
+LOG = logging.getLogger(__name__)
+STATE_MACHINE_ARN = "stateMachineArn"
+EXECUTION_ARN = "executionArn"
+INPUT = "input"
+RUNNING = "RUNNING"
+SFN_EXECUTION_WAIT_TIME = 2
+
+
+class StepFunctionsStartExecutionExecutor(BotoActionExecutor):
+    """
+    Calls "start_execution" method of "stepfunctions" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _stepfunctions_client: Any
+    _state_machine_arn: str
+    _remote_output_format: RemoteInvokeOutputFormat
+    request_parameters: dict
+
+    def __init__(self, stepfunctions_client: Any, physical_id: str, remote_output_format: RemoteInvokeOutputFormat):
+        self._stepfunctions_client = stepfunctions_client
+        self._remote_output_format = remote_output_format
+        self._state_machine_arn = physical_id
+        self.request_parameters = {}
+
+    def validate_action_parameters(self, parameters: dict):
+        for parameter_key, parameter_value in parameters.items():
+            if parameter_key == "stateMachineArn":
+                LOG.warning("stateMachineArn is defined using the value provided for resource_id argument.")
+            elif parameter_key == "input":
+                LOG.warning("input is defined using the value provided for either --event or --event-file options.")
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+        if not self.request_parameters.get("name"):
+            timestamp = str(int(time.time() * 1000))
+            self.request_parameters["name"] = f"sam_remote_invoke_{timestamp}"
```

Comment:
nit: timestamp is nice, iso formatted string would be even better for readability